lib/CL/devices/devices.c

```diff
@@ -206,7 +206,7 @@ pocl_init_devices()
      infinite loop. */
 
   if (init_in_progress)
-      return;
+      return CL_SUCCESS; /* debatable, but what else can we do ? */
```

Comment:
This is here because of recursive initialization that happens due to hwloc's OpenCL plugin: When we initialize hwloc, it initializes OpenCL which can be pocl, etc.